cpp/src/gandiva/llvm_generator.cc

```diff
@@ -832,6 +832,46 @@ void LLVMGenerator::Visitor::Visit(const BooleanOrDex& dex) {
   result_.reset(new LValue(result_value));
 }
 
+#define MAKE_VISIT_IN(ctype)                                                           \
```

Comment:
I'm not sure why you define a macro for this rather than a private templatized helper func. Long macros are tedious to maintain and debug.

Reply:
i plead ignorance :( did not know functions could be templatized too..
changed it to a template helper function.